include/geolib/ros/msg_conversions.h

```diff
@@ -42,6 +45,31 @@ inline void convert(const geo::Transform& t, geometry_msgs::Transform& msg) {
     convert(t.getBasis(), msg.rotation);
 }
 
+inline void convert(const TriangleI& t, shape_msgs::MeshTriangle& msg) {
+    msg.vertex_indices[0] = t.i1_;
+    msg.vertex_indices[1] = t.i2_;
+    msg.vertex_indices[2] = t.i3_;
+}
+
+void convert(const geo::Mesh& m, shape_msgs::Mesh& msg) {
```

Comment:
Also misses documentation.

I agree it's too big for inlining. Having this function here however means it gets inserted into the program by each compilation unit (each .cpp file) that directly or indirectly includes this file. At link-time, you will run into "multiple definitions" problems, since you cannot have the same function with the same signature several times in one program.
This implementation should move to a .cpp file instead, and here just have the function declaration. (Documentation should go near the implementation to keep code and documentation close to each other, increasing the odds it stays in sync.)